File: git_push.py
#!/usr/bin/env python3
import subprocess
import os
import sys

os.chdir('c:\\PROJECT')

# Set environment to avoid ssh-keygen
env = os.environ.copy()
env['GIT_SSH_COMMAND'] = 'ssh -o StrictHostKeyChecking=no'
env['GIT_ASKPASS'] = 'c:\\PROJECT\\askpass.bat'

# Configuration
commands = [
    ['git', 'config', '--global', 'credential.helper', 'wincred'],
    ['git', 'config', '--global', 'user.name', 'Silentt03'],
    ['git', 'config', '--global', 'user.email', 'unioncollective03@gmail.com'],
    ['git', 'remote', 'remove', 'origin'],
    ['git', 'remote', 'add', 'origin', 'https://github.com/Silentt03/Aplikasi-Stock.git'],
    ['git', 'status'],
    ['git', 'remote', '-v'],
]

print("="*60)
print("Git Push Automation Script")
print("="*60)

for cmd in commands:
    print(f"\n$ {' '.join(cmd)}")
    try:
        result = subprocess.run(cmd, capture_output=True, text=True, env=env, timeout=10)
        if result.stdout:
            print(result.stdout.strip())
        if result.stderr and 'warning' not in result.stderr.lower():
            print("Error:", result.stderr.strip())
    except Exception as e:
        print(f"Error executing command: {e}")

# Now attempt the push
print("\n" + "="*60)
print("Attempting to push to GitHub...")
print("="*60)

push_cmd = ['git', 'push', '-u', 'origin', 'main']
print(f"$ {' '.join(push_cmd)}")

try:
    result = subprocess.run(push_cmd, capture_output=True, text=True, env=env, timeout=30, input='\n\n')
    if result.stdout:
        print(result.stdout.strip())
    if result.returncode == 0:
        print("\n✅ Push successful!")
    else:
        print(f"\n❌ Push failed with return code {result.returncode}")
        if result.stderr:
            print("Error output:")
            print(result.stderr.strip())
except subprocess.TimeoutExpired:
    print("❌ Command timed out")
except Exception as e:
    print(f"❌ Error: {e}")

print("\n" + "="*60)

File: git_push.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const projectDir = 'c:\\PROJECT';
process.chdir(projectDir);

console.log('='.repeat(60));
console.log('Git Push Automation - Node.js Version');
console.log('='.repeat(60));

try {
    console.log('\n1. Killing ssh-keygen processes...');
    try {
        execSync('taskkill /F /IM ssh-keygen.exe', { stdio: 'pipe' });
        console.log('   ✓ Processes killed');
    } catch (e) {
        console.log('   - No ssh-keygen processes found (OK)');
    }

    console.log('\n2. Configuring git...');
    execSync('git config --global user.name "Silentt03"', { stdio: 'pipe' });
    console.log('   ✓ User name set');
    
    execSync('git config --global user.email "unioncollective03@gmail.com"', { stdio: 'pipe' });
    console.log('   ✓ User email set');
    
    execSync('git config --global credential.helper wincred', { stdio: 'pipe' });
    console.log('   ✓ Credential helper set');

    console.log('\n3. Checking git status...');
    const status = execSync('git status --short', { encoding: 'utf-8' });
    if (status.trim()) {
        console.log('   Changes to commit:');
        console.log(status);
    } else {
        console.log('   ✓ Working tree clean');
    }

    console.log('\n4. Checking remote configuration...');
    const remote = execSync('git remote -v', { encoding: 'utf-8' });
    console.log(remote);

    console.log('\n5. Attempting to push to GitHub...');
    console.log('   $ git push -u origin main');
    
    try {
        const pushResult = execSync('git push -u origin main 2>&1', { 
            encoding: 'utf-8',
            stdio: ['pipe', 'pipe', 'pipe'],
            timeout: 30000
        });
        console.log(pushResult);
        console.log('\n✅ Push successful!');
    } catch (pushError) {
        const output = pushError.stdout ? pushError.stdout.toString() : '';
        const stderr = pushError.stderr ? pushError.stderr.toString() : '';
        
        console.log('Output:', output);
        if (stderr) console.log('Error:', stderr);
        
        if (pushError.code !== 0) {
            console.log('\n⚠️ Push encountered an error.');
            console.log('This might be due to:');
            console.log('  - Authentication failure (credentials invalid)');
            console.log('  - Network issue');
            console.log('  - Repository access denied');
            console.log('\nTry pushing manually with: git push -u origin main');
        }
    }

} catch (error) {
    console.error('\n❌ Error:', error.message);
    if (error.stdout) console.log('Output:', error.stdout.toString());
}

console.log('\n' + '='.repeat(60));
console.log('Done!');
console.log('='.repeat(60));
